plugins/github-deployments/src/components/GithubDeploymentsCard.tsx

```diff
@@ -55,7 +57,13 @@ const GithubDeploymentsComponent = ({
   );
 };
 
-export const GithubDeploymentsCard = ({ last }: { last?: number }) => {
+export const GithubDeploymentsCard = ({
+  last,
+  host,
+}: {
+  last?: number;
+  host?: string;
```

Comment:
To be honest it's a bit of a drag to have to set the host here, and doesn't handle components mixed in from multiple providers well.

I think it would be worth investigating if you can use the `backstage.io/source-location` annotation, and if that doesn't exist fall back to the `backstage.io/managed-by-location` to figure out the location of the source code, and by that what host the component is using.

Reply:
> To be honest it's a bit of a drag to have to set the host here, and doesn't handle components mixed in from multiple providers well.
> 
> I think it would be worth investigating if you can use the `backstage.io/source-location` annotation, and if that doesn't exist fall back to the `backstage.io/managed-by-location` to figure out the location of the source code, and by that what host the component is using.

Hey, thanks for the feedback.

Agree it's a drag to be passing the host - however I believe this would currently accept multiple providers but they'd need to be added to the integration list and the consumer would need to do some host handling themselves.

If I'm understanding it correctly, your suggestion would look something like:

```typescript
const location = entity?.metadata.annotations?.['backstage.io/source-location']
    || entity?.metadata.annotations?.['backstage.io/managed-by-location'];

  const apiBaseUrl = scmIntegrationsApi.github.list()
    .find(integration => location.includes(integration.config.host))?.config?.apiBaseUrl;
```

Reply:
@anderoo Yep, pretty much! Although you should be able to use this to get the matching config more directly:

```ts
scmIntegrationsApi.github.byUrl(parseLocationReference(location).target)
```

Reply:
@Rugvip Thanks -- I've updated code and tests, let me know if that's what you intended.

Reply:
Need to figure out how to default this correctly. Potentially by ensuring type == 'url'.

Reply:
Updated to handle cases where location != url type.